lib/common/tests/schemas/get_schema_version_test.c

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2023 the Pacemaker project contributors
+ *
+ * The version control history for this file may have further details.
+ *
+ * This source code is licensed under the GNU General Public License version 2
+ * or later (GPLv2+) WITHOUT ANY WARRANTY.
+ */
+
+#include <crm_internal.h>
+
+#include <crm/msg_xml.h>
+#include <crm/common/unittest_internal.h>
+#include <crm/common/xml_internal.h>
+
+static int
+setup(void **state) {
+    setenv("PCMK_schema_directory", PCMK__TEST_SCHEMA_DIR, 1);
+    crm_schema_init();
+    return 0;
+}
+
+static int
+teardown(void **state) {
+    crm_schema_cleanup();
+    unsetenv("PCMK_schema_directory");
+    return 0;
+}
+
+static void
+bad_input(void **state) {
+    assert_int_equal(16, get_schema_version(NULL));
+    assert_int_equal(-1, get_schema_version(""));
+    assert_int_equal(-1, get_schema_version("blahblah"));
+    assert_int_equal(-1, get_schema_version("pacemaker-2.47"));
+    assert_int_equal(-1, get_schema_version("pacemaker-47.0"));
+}
+
+static void
+typical_usage(void **state) {
+    assert_int_equal(0, get_schema_version("pacemaker-1.0"));
+    assert_int_equal(1, get_schema_version("pacemaker-1.2"));
+    assert_int_equal(3, get_schema_version("pacemaker-2.0"));
+    assert_int_equal(8, get_schema_version("pacemaker-2.5"));
+    assert_int_equal(14, get_schema_version("pacemaker-3.0"));
```

Comment:
I notice `get_schema_version( )` uses `pcmk__str_casei`. It probably shouldn't, but for backward compatibility we should keep it (it's used with the CIB `validate-with` attribute). We might as well test it.

Reply:
Updated with uppercase versions.